### PR TITLE
Add -OrderBy combinator

### PR DESCRIPTION
**Pull request by Sonichka1311**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

Changelog category (leave one):
- New Feature

Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Add -OrderBy combinator && change transformArguments signature

### CODE REVIEW

**Comment by akuzm** [comment]
Please add a brief description of what it does to https://github.com/ClickHouse/ClickHouse/blob/master/docs/en/sql-reference/aggregate-functions/combinators.md

File: src/AggregateFunctions/AggregateFunctionOrderBy.h [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+    }
+
+    bool hasTrivialDestructor() const override
+    {
+        return nested_func->hasTrivialDestructor();
+    }
+
+    void add(AggregateDataPtr place, const IColumn ** columns, size_t row_num, Arena * arena) const override
+    {
+        std::call_once(data(place).initialized, [this, place, columns]() {
```
Comment (akuzm) [inline src/AggregateFunctions/AggregateFunctionOrderBy.h:103]: `add` is not supposed to be called concurrently on shared data, so the `call_once` shouldn't be necessary. Or is it?

**Comment by hczhcz** [comment]
It is probably good to name it as `-Sort` considering we are using different terms in SQL functions and statements (sort vs order by, if vs where/having, uniq vs distinct, ...)

File: tests/queries/0_stateless/01281_combinator_orderby.sql [hunk 2]
```diff
@@ -0,0 +1,3 @@
+SELECT groupArrayOrderBy(2)(x, y, z) FROM (SELECT number AS x, number % 3 AS y, number % 5 AS z FROM system.numbers LIMIT 10);
+SELECT groupArrayOrderBy(1)(x, y) FROM (SELECT number AS x, number % 3 AS y FROM system.numbers_mt LIMIT 10);
+SELECT groupArrayOrderBy(1)(x, y) FROM (SELECT number AS x, number AS y FROM system.numbers_mt LIMIT 10);
```
Comment (qoega) [inline tests/queries/0_stateless/01281_combinator_orderby.sql:3]: 1) Lets add some other examples if applicable - not only `groupArray`.
2) Negative cases are also useful. Both incorrect column names, and unsupported functions.

**Comment by filimonov** [comment]
https://github.com/ClickHouse/ClickHouse/issues/3708

File: src/AggregateFunctions/AggregateFunctionOrderBy.h [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+
+struct AggregateFunctionOrderByData
+{
+    AggregateFunctionOrderByData() = default;
+    explicit AggregateFunctionOrderByData(size_t num_arguments): value(num_arguments), arena(nullptr) {}
+
+    using Array = std::vector<IColumn*>;
+
+    Array value;
+    std::once_flag initialized;
```
Comment (alexey-milovidov) [inline src/AggregateFunctions/AggregateFunctionOrderBy.h:26]: The proposed variant of implementation is not the correct way to implement this task.

**Comment by alexey-milovidov** [comment]
We can implement it again in the following way:
- the function with OrderBy combinator will have state like `groupArray` and collect pairs of value and order by fields;
- only at the finalization step the collected array will be fed to the transformed aggregate function.

**Pull request closed without merging**